src/events.interfaces.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    Callback,
+    Dictionary,
+    SDKEventAttrs,
+    SDKEventOptions,
+    TransactionAttributes,
+} from '@mparticle/web-sdk';
+import {
+    BaseEvent,
+    SDKEventCustomFlags,
+    SDKProduct,
+    SDKProductImpression,
+    SDKPromotion,
+} from './sdkRuntimeModels';
+import { valueof } from './utils';
+import { EventType, ProductActionType, PromotionActionType } from './types';
+
+type dataFunction = (element: HTMLElement) => Dictionary<string>;
+type dataObject = Dictionary<string>;
+
+export interface IEvents {
+    addEventHandler(
+        domEvent: string,
+        selector: string | Node,
+        eventName: string,
+
+        // QUESTION: In what cases would data be a function?
+        data: dataObject | dataFunction,
```

Comment:
This predates me all the way to the very first commit of the SDK. 
https://github.com/mParticle/mparticle-web-sdk/commit/050d15a1e9bf8f6a04b883ecc4c8404edbd00ee8#diff-95045420909226e4347b6b716fe6ae6fdfb25642d6571953b756744578d93180R657-R730

Unclear to me why a function would be passed.  Nothing in docs about this.  Bonus points if you can figure out why :)

Reply:
So I was able to do some digging and this was intended to allow customers to wrap event handlers and pass them in as callbacks to `logEvent`. As we don't have this documented anywhere, I wonder if it's something that is even used. I think we should consider deprecating this function unless there's a good reason to keep it.

From what I can tell, we don't even have any tests around this.